idb/postgres/postgres_migrations.go

```diff
@@ -2204,3 +2205,32 @@ func ClearAccountDataMigration(db *IndexerDb, state *MigrationState) error {
 
 	return nil
 }
+
+// MakeDeletedNotNullMigration makes "deleted" columns NOT NULL in tables
+// account, account_asset, asset, app, account_app.
+func MakeDeletedNotNullMigration(db *IndexerDb, state *MigrationState) error {
+	db.accountingLock.Lock()
+	defer db.accountingLock.Unlock()
+
+	queries := []string{
+		"UPDATE account SET deleted = false WHERE deleted is NULL",
+		"ALTER TABLE account ALTER COLUMN deleted SET NOT NULL",
```

Comment:
I think this isn't quite safe. There could be a moment between `UPDATE` and `ALTER TABLE` where a new row is added but it doesn't have a value set so is NULL which violates NOT NULL.
But, we can
```sql
ALTER TABLE account ALTER COLUMN deleted SET DEFAULT false
```
Then new rows will get the default value written. _Then_ `UPDATE`, then `ALTER ... SET NOT NULL`

Reply:
Do you mean somebody can connect to DB and manually insert a row? We do use the accounting mutex, so the accounting code will not create conflicts.

Reply:
I guess that makes it safe.

Reply:
In addition to that, it would require (a) multiple writers, and (b) one of the writers would have to be an old version.

I think our bases are covered here even if we didn't have the mutex.